reTurn/AsyncTcpSocketBase.cxx

```diff
@@ -76,26 +75,17 @@ AsyncTcpSocketBase::handleTcpResolve(const asio::error_code& ec,
 
 void 
 AsyncTcpSocketBase::handleConnect(const asio::error_code& ec,
-                                  asio::ip::tcp::resolver::iterator endpoint_iterator)
+                                  const asio::ip::tcp::endpoint& endpoint)
 {
    if (!ec)
    {
       // The connection was successful.
       mConnected = true;
-      mConnectedAddress = endpoint_iterator->endpoint().address();
-      mConnectedPort = endpoint_iterator->endpoint().port();
+      mConnectedAddress = endpoint.address();
+      mConnectedPort = endpoint.port();
 
       onConnectSuccess();
    }
-   else if (++endpoint_iterator != asio::ip::tcp::resolver::iterator())
```

Comment:
Was this else block intentionally removed?

Reply:
Yes. It restarted the `async_connect` to another remote address after incrementing the iterator. This is handled by the standalone `asio::async_connect` algorithm, which takes a list of endpoints and passes the endpoint that got connected to the callback.